### PR TITLE
Allow for SPN context persistence in .NET Core

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Fix for https://github.com/Azure/azure-powershell/issues/6743 

Common packages based on PR https://github.com/Azure/azure-powershell-common/pull/32

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/1271/
.NET Core sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/185/

File: src/ResourceManager/Profile/Commands.Profile/Account/ConnectAzureRmAccount.cs [hunk 1]
```diff
@@ -269,6 +269,13 @@ public override void ExecuteCmdlet()
                 azureAccount.SetProperty(AzureAccount.Property.Tenants, new[] { TenantId });
             }
 
+#if NETSTANDARD
+            if (azureAccount.Type == AzureAccount.AccountType.ServicePrincipal)
+            {
+                azureAccount.SetProperty(AzureAccount.Property.ServicePrincipalSecret, password.ConvertToString());
+            }
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Account/ConnectAzureRmAccount.cs]: write warning if modificationscope is currentuser

File: src/ResourceManager/Profile/Commands.Profile/packages.config [hunk 2]
```diff
@@ -10,23 +10,23 @@
...
-  <package id="Microsoft.Azure.PowerShell.KeyVault" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.Network" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.ResourceManager" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.ScenarioTest.ResourceManager" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.ServiceManagement" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.Storage" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.Storage.Management" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.Strategies" version="1.1.0-preview" targetFramework="net452" />
-  <package id="Microsoft.Azure.PowerShell.Websites" version="1.1.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.PowerShell.Aks" version="1.0.120-preview" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/packages.config]: Will need to update versions

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/1326/
.NET Core sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/191/

File: src/ResourceManager/Profile/Commands.Profile/Account/ConnectAzureRmAccount.cs [hunk 3]
```diff
@@ -269,6 +269,20 @@ public override void ExecuteCmdlet()
...
 
+#if NETSTANDARD
+            if (azureAccount.Type == AzureAccount.AccountType.ServicePrincipal)
+            {
+                azureAccount.SetProperty(AzureAccount.Property.ServicePrincipalSecret, password.ConvertToString());
+                if (GetContextModificationScope() == ContextModificationScope.CurrentUser)
+                {
+                    var file = AzureSession.Instance.ARMProfileFile;
+                    var directory = AzureSession.Instance.ARMProfileDirectory;
+                    WriteWarning("The provided service principal secret will be included in the " + file + " file found in the user profile " +
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Account/ConnectAzureRmAccount.cs]: This message needs to be in the resources file.  I like the text

**Comment by cormacpayne** [comment]
Closing in favor of https://github.com/Azure/azure-powershell/pull/7289

**Pull request closed without merging**